ethsigner/signer/azure/src/main/java/tech/pegasys/ethsigner/signer/azure/AzureKeyVaultTransactionSignerFactory.java

```diff
@@ -40,23 +40,23 @@
   private static final Logger LOG = LogManager.getLogger();
 
   private static final String AZURE_URL_PATTERN = "https://%s.vault.azure.net";
-
-  private final KeyVaultClientCustom client;
-  private final String baseUrl;
+  private final AzureKeyVaultAuthenticator vaultAuthenticator;
 
   public AzureKeyVaultTransactionSignerFactory(
-      final String keyVaultName, final KeyVaultClientCustom client) {
-    this.client = client;
-    this.baseUrl = constructAzureKeyVaultUrl(keyVaultName);
+      final AzureKeyVaultAuthenticator vaultAuthenticator) {
+    this.vaultAuthenticator = vaultAuthenticator;
   }
 
-  public TransactionSigner createSigner(final String keyName, final String keyVersion) {
-    checkNotNull(keyName, "keyName must be specified");
+  public TransactionSigner createSigner(final AzureConfig config) {
+    checkNotNull(config, "Config must be specified");
+    final KeyVaultClientCustom client =
+        vaultAuthenticator.getAuthenticatedClient(config.getClientId(), config.getClientSecret());
+    final String baseUrl = constructAzureKeyVaultUrl(config.getKeyvaultName());
```

Comment:
nit: we have both Keyvault and KeyVault

Reply:
not sure which way to go - is keyvault one, or two words? Or do we just want to call it a vault? (and the key is implied)

Reply:
but yes going with "vault" also solves the problem